source

```
@@ -62535,6 +62540,49 @@ try {
+    <var>radii</var> represents a list of radiuses for the corners of the rectangle, in pixels. The
+    number and order of these radii function in the same way as the CSS <span>'border-radius'</span>
+    property.</p>
+
+    <p>If <var>w</var> and <var>h</var> are both greater than or equal to 0, or if both are smaller
+    than 0, then the path is drawn clockwise. Otherwise, it is drawn counterclockwise.</p>
+
+    <p>When <var>w</var> is negative, the rounded rectangle is flipped horizontally, which means
+    that the radius values that normally apply to the left corners are used on the right and vice
+    versa. Similarly, when <var>h</var> is negative, the rounded rect is flipped vertically.</p>
+
+    <p>When a value <var>r</var> in <var>radii</var> is a number, the corresponding corner(s) are
+    drawn as circular arcs of radius <var>r</var>.</p>
+
+    <p>When a value <var>r</var> in <var>radii</var> is an object with <code data-x="">{ x, y
+    }</code> properties, the corresponding corner(s) are drawn as elliptical arcs whose <var>x</var>
+    and <var>y</var> radii are equal to <var>r.x</var> and <var>r.y</var>, respectively.</p>
+
+    <p>When the sum of the <var>radii</var> of two corners of the same edge is greater than the
+    length of the edge, then all the <var>radii</var> of the rounded rectangle are scaled by a
+    factor of length / (<var>r1</var> + <var>r2</var>). If multiple edges have this property, the
+    scale factor of the edge with the smallest scale factor is used. This is consistent with CSS
+    behavior.</p>
+
+    <p>Throws a <code data-x="js-RangeError">RangeError</code> if a value in <var>radii</var> is a
+    negative number, or is an <code data-x="">{ x, y }</code> object whose <code data-x="">x</code>
+    or <code data-x="">y</code> properties are negative numbers.</p>
+
+    <!-- Consider adding this useful information to all the methods which behave this way: -->
```

I commented this out for now and added a comment above explaining that we should consider adding it everywhere. I think it is helpful information that would be good to have but I worry that only one method explaining when it silently does nothing would be confusing for developers.

If you're up for fixing the other methods in this PR or a followup PR, that'd be lovely :).

---

So you mean all path functions that fail silently and return if their inputs are non-finite?

I'd be happy to tackle that in a separate PR once the Canvas2D new api is launched 😃